docs/REST-API/18-PCL-Overview.md

```diff
@@ -0,0 +1,590 @@
+```
+
+If the two times are the same, that indicates a 24 hour period.
+
+```
+# From Sat noon through Mon noon using Cairo time
+# (Currently Cairo is on EGY time year round & does not observe daylight saving time)
+Sat,Sun 12:00:00 to 12:00:00 Africa/Cairo
+```
+
+The allowable timezones are defined in the [tz database][1].
+
+When doing comparisons of times, PCL compares the times according to what a clock on the wall would say in that timezone. That is, local daylight saving time is factored in. This can cause slightly unexpected behavior during the times when the clock changes from daylight saving time to standard time and vice versa.
+
+For example, when daylight saving time ends, the clocks are changed from 2:00 AM back to 1:00 AM. This means that a wall clock, will repeat the 1:00 AM to 2:00 AM hour that day. Considering a PCL condition that evaluates to `true` from 1:30am through 3:00 AM on Sundays in New York, when the clocks rollback to 1:00 AM, then the half hour between 1:00 AM and 1:30 AM produces a gap in the schedule. The following table shows the evaluation at various points in time around when DST ends:
+
+**PCL Schedule: `Sun 01:30:00 to 03:15:00 America/New_York`**
+
+Wall Clock Time | UTC Time | Matches Schedule? | Notes
+-- | -- | -- | --
+[01:00 EDT (UTC-0400)](https://www.timeanddate.com/worldclock/converter.html?iso=20211107T050000&p1=179&p2=1440) | 05:00 UTC | ❌ No |
+01:30 EDT (UTC-0400) | 05:30 UTC | ✅ Yes |
+DST ends: [01:00 EST (UTC-0500)](https://www.timeanddate.com/worldclock/converter.html?iso=20211107T060000&p1=179&p2=1440) | 06:00 UTC | ❌ No | This is the gap
+01:15 EST (UTC-0500) | 06:15 UTC | ❌ No | This is the gap
+01:30 EST (UTC-0500) | 06:30 UTC | ✅ Yes |
+02:00 EST (UTC-0500) | 07:00 UTC | ✅ Yes |
+02:30 EST (UTC-0500) | 07:30 UTC | ✅ Yes |
+03:00 EST (UTC-0500) | 08:00 UTC | ✅ Yes |
+03:30 EST (UTC-0500) | 08:30 UTC | ❌ No |
+
+Another example similar to the above, but when DST starts.
+
+**PCL Schedule: `Sun 01:30:00 to 03:15:00 America/New_York`**
+
+Wall Clock Time | UTC Time | Matches Schedule? | Notes
+-- | -- | -- | --
+00:00 EST (UTC-0500) | 05:00 UTC | ❌ No |
+00:30 EST (UTC-0500) | 05:30 UTC | ❌ No |
+01:00 EST (UTC-0500) | 06:00 UTC | ❌ No |
+01:30 EST (UTC-0500) | 06:30 UTC | ✅ Yes |
+01:59 EST (UTC-0500) | 06:59 UTC | ✅ Yes |
+DST begins: 03:00 EDT (UTC-0400) | 07:00 UTC | ✅ Yes | 2:00 AM skipped to 3:00 AM
+03:15 EDT (UTC-0400) | 07:15 UTC | ✅ Yes |
+03:30 EDT (UTC-0400) | 07:30 UTC | ❌ No |
+04:00 EDT (UTC-0400) | 08:00 UTC | ❌ No |
+04:30 EDT (UTC-0400) | 08:30 UTC | ❌ No |
+
+#### Nil
+
+A `nil` value represents the absense of any other value type.
+
+### Built-in Operations
+
+PCL has a variety of built-in operations:
+- boolean operations: `and`, `or`, `not`
+- numeric comparison operations: `>`, `>=`, `<`, `<=`
+- string comparison operations: `matches [exactly]`, `matches part [exactly]`, `matches regex [exactly]`
+- other operations: `==`, `in`, `exists`, `now`
```

Comment:
**Not a blocker**: We should probably mention _somewhere_ in this doc that some of the conditions (like `now >` and this one) are only available to EI customers, and maybe link to https://support.pagerduty.com/docs/event-intelligence

Reply:
Is it just `now`, `trigger_count`, and `resetting_trigger_count` that are only available to EI customer?

Reply:
Yup! Exactly those. (And to be clear, restricts use of `now` for both for numeric _and_ schedule operators)